src/components/SearchInput/index.jsx

```diff
@@ -31,32 +29,67 @@ function SearchInput() {
       }
     }
 
-    getAutoCompletions(userInput);
-  }, [userInput]);
+    if (!arrowKeyPressed.current) {
+      getAutoCompletions(userInput);
+    }
+  }, [userInput, arrowKeyPressed]);
 
-  function handleArrowKeyPress(event) {
+  function handleUserInputChange(event) {
+    event.preventDefault();
+
+    arrowKeyPressed.current = false;
+
+    setUserInput(event.target.value);
+    setShowAutoCompletions(true);
+    setSelectedItemIndex(-1);
+  }
+
+  function handleKeyPress(event) {
     const pressedKey = event.key;
 
-    if (
-      pressedKey !== "ArrowUp" &&
-      pressedKey !== "ArrowDown" &&
-      pressedKey !== "Escape"
-    ) {
-      return;
-    }
+    if (pressedKey === "ArrowDown" || pressedKey === "ArrowUp") {
+      arrowKeyPressed.current = true;
 
-    event.preventDefault();
+      if (!showAutoCompletions) {
+        return setShowAutoCompletions(true);
+      }
+    }
 
     if (pressedKey === "ArrowDown") {
       setSelectedItemIndex((prevIndex) =>
         prevIndex < autoCompletions.length - 1 ? prevIndex + 1 : 0,
       );
+      setUserInput(
+        autoCompletions[
+          selectedItemIndex < autoCompletions.length - 1
+            ? selectedItemIndex + 1
+            : 0
+        ],
+      );
```

Comment:
```suggestion
      const index = selectedItemIndex < autoCompletions.length - 1
            ? selectedItemIndex + 1
            : 0
      setUserInput(
        autoCompletions[index],
      );
```
배열 인덱스 안에 삼항연산자보다는 밖에서 변수로 빼는게 가독성에 좋아 보여요